### PR TITLE
Invalidate guards

### PR DESCRIPTION
**Pull request by d7919**
Introduces `invalidateGuards` function to set guard cells to Nan for Field3D and Field2D. This is only enabled for CHECK>2. This routine is automatically called when allocating a new block of memory in Field3D/Field2D allocate (but not when making a unique block from existing data).

This can be helpful to catch uses of uninitialised guard values (e.g. for derivative operations). This has already caught some errors and this PR includes some "fixes" in the input files for the integrated tests, mostly in Laplacian treatments. We would like to be able to revert this commit which can be achieved by fixing the framework.

### CODE REVIEW

File: tests/unit/field/test_field2d.cxx [hunk 1]
```diff
@@ -447,6 +448,49 @@ TEST_F(Field2DTest, CheckData) {
...
+
+  // Count the number of non-boundary points
+  sum = 0;
+  for (const auto &i : field.region(RGN_NOBNDRY)) {
+    field[i] = 0.0; // Reset field value
+    sum++;
+  }
+  const int nbndry = nmesh - sum;
+
+#if CHECK > 2
```
Comment (bendudson) [inline tests/unit/field/test_field2d.cxx]: What check level is the test suite run at?

Comment (d7919) [reply tests/unit/field/test_field2d.cxx]: It depends on the user's configure choices (e.g. the travis builds includes CHECK=0, CHECK=3 and the default of CHECK=2 [I think]).

Comment (bendudson) [reply tests/unit/field/test_field2d.cxx]: ok, just checking that this is tested on Travis. Thanks!

File: tests/integrated/test-gyro/data/BOUT.inp [hunk 2]
```diff
@@ -21,3 +21,6 @@ symmetricGlobalY = false
 
 [output]
 floats = true
+
+[laplace]
+all_terms = false
```
Comment (dschwoerer) [inline tests/integrated/test-gyro/data/BOUT.inp]: missing newline

**Pull request merged**